crates/ruma-common/src/events/room/message.rs

```diff
@@ -498,17 +508,17 @@ impl InReplyTo {
 /// [replaces another event]: https://spec.matrix.org/v1.4/client-server-api/#event-replacements
 #[derive(Clone, Debug)]
 #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
-pub struct Replacement {
+pub struct Replacement<C> {
     /// The ID of the event being replaced.
     pub event_id: OwnedEventId,
 
     /// New content.
-    pub new_content: Box<RoomMessageEventContent>,
+    pub new_content: Box<C>,
```

Comment:
With this change we are no longer nesting the event content inside itself, I think we can drop this `Box`.